patches/roslyn/0002-Add-VBCSCompiler-to-SourceBuild.sln.patch

```diff
@@ -0,0 +1,54 @@
+From da00e2f1d9f24eaf33459cfea42677b0f887edfb Mon Sep 17 00:00:00 2001
+From: Davis Goodin <dagood@microsoft.com>
+Date: Mon, 16 Apr 2018 15:16:02 -0500
+Subject: [PATCH] Add VBCSCompiler to SourceBuild.sln
+
+---
+ SourceBuild.sln | 18 ++++++++++++++++++
+ 1 file changed, 18 insertions(+)
+
+diff --git a/SourceBuild.sln b/SourceBuild.sln
+index f7a5c2d41a..5f914cce20 100644
+--- a/SourceBuild.sln
++++ b/SourceBuild.sln
+@@ -24,6 +24,10 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "vbc", "src\Compilers\Visual
+ EndProject
+ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "MSBuildTask", "src\Compilers\Core\MSBuildTask\MSBuildTask.csproj", "{7AD4FE65-9A30-41A6-8004-AA8F89BCB7F3}"
+ EndProject
++Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "VBCSCompiler", "src\Compilers\Server\VBCSCompiler\VBCSCompiler.csproj", "{649CE05A-529C-40D7-A501-CA9BCFEE40DC}"
++EndProject
++Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Server", "Server", "{B57E7F86-EB16-4F60-A47C-F9D170F54400}"
++EndProject
+ Global
+ 	GlobalSection(SolutionConfigurationPlatforms) = preSolution
+ 		Debug|Any CPU = Debug|Any CPU
+@@ -106,6 +110,18 @@ Global
+ 		{7AD4FE65-9A30-41A6-8004-AA8F89BCB7F3}.Release|x64.Build.0 = Release|x64
+ 		{7AD4FE65-9A30-41A6-8004-AA8F89BCB7F3}.Release|x86.ActiveCfg = Release|x86
+ 		{7AD4FE65-9A30-41A6-8004-AA8F89BCB7F3}.Release|x86.Build.0 = Release|x86
++		{649CE05A-529C-40D7-A501-CA9BCFEE40DC}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
++		{649CE05A-529C-40D7-A501-CA9BCFEE40DC}.Debug|Any CPU.Build.0 = Debug|Any CPU
++		{649CE05A-529C-40D7-A501-CA9BCFEE40DC}.Debug|x64.ActiveCfg = Debug|Any CPU
++		{649CE05A-529C-40D7-A501-CA9BCFEE40DC}.Debug|x64.Build.0 = Debug|Any CPU
++		{649CE05A-529C-40D7-A501-CA9BCFEE40DC}.Debug|x86.ActiveCfg = Debug|Any CPU
++		{649CE05A-529C-40D7-A501-CA9BCFEE40DC}.Debug|x86.Build.0 = Debug|Any CPU
++		{649CE05A-529C-40D7-A501-CA9BCFEE40DC}.Release|Any CPU.ActiveCfg = Release|Any CPU
++		{649CE05A-529C-40D7-A501-CA9BCFEE40DC}.Release|Any CPU.Build.0 = Release|Any CPU
++		{649CE05A-529C-40D7-A501-CA9BCFEE40DC}.Release|x64.ActiveCfg = Release|Any CPU
++		{649CE05A-529C-40D7-A501-CA9BCFEE40DC}.Release|x64.Build.0 = Release|Any CPU
++		{649CE05A-529C-40D7-A501-CA9BCFEE40DC}.Release|x86.ActiveCfg = Release|Any CPU
++		{649CE05A-529C-40D7-A501-CA9BCFEE40DC}.Release|x86.Build.0 = Release|Any CPU
+ 	EndGlobalSection
+ 	GlobalSection(NestedProjects) = preSolution
+ 		{DDFAE591-7414-4CC9-AD99-427C2D153893} = {6D173F59-562E-4BAA-8D98-9A8F4901D99A}
+@@ -118,5 +134,7 @@ Global
+ 		{E3CD2895-76A8-4D11-A316-EA67CB5EA42C} = {9EECE6A3-C098-44D0-8ACD-0D8691EC8C3F}
+ 		{8CE3A581-2969-4864-A803-013E9D977C3A} = {11F63FDD-6EB7-4993-8769-8C8DC5DA29C3}
+ 		{7AD4FE65-9A30-41A6-8004-AA8F89BCB7F3} = {5DC491BA-E836-47C1-A4AA-3C770742718B}
++		{649CE05A-529C-40D7-A501-CA9BCFEE40DC} = {B57E7F86-EB16-4F60-A47C-F9D170F54400}
++		{B57E7F86-EB16-4F60-A47C-F9D170F54400} = {DDFAE591-7414-4CC9-AD99-427C2D153893}
+ 	EndGlobalSection
+ EndGlobal
```

Comment:
Why is this patch being added? No patches should be needed for Roslyn.

Reply:
I had to add it at the time so that the files `Microsoft.NETCore.Compilers` needed were available. I'll try this build locally without the patch in case it's no longer needed with the recent PR.

Reply:
This shouldn't be necessary. @333fred recently sent a PR that moved to us building all of our NuGet packages without any patches.

Reply:
Note that I'm adding the build/pack of `Microsoft.NETCore.Compilers`, so this isn't surprising to me.

The first error I hit is:

```
.../Tools/dotnetcli/sdk/2.1.101/Sdks/NuGet.Build.Tasks.Pack/build/NuGet.Build.Tasks.Pack.targets(194,5): error : Could not find a part of the path '.../src/roslyn/Binaries/Release/Dlls/MSBuildTask/netcoreapp2.0/publish/runtimes'.
```

That led me to add the `Exec` that runs `/t:PublishWithoutBuilding`, mimicking the behavior of https://github.com/dotnet/roslyn/blob/94e73792cfec399f8e1ab3e9246703cbddabbcd4/build/scripts/build.ps1#L280-L288. But running *just* that, I get:

```
.../Tools/dotnetcli/sdk/2.1.101/Sdks/Microsoft.NET.Sdk/build/Microsoft.PackageDependencyResolution.targets(327,5): error : Assets file '.../src/roslyn/Binaries/Obj/VBCSCompiler/project.assets.json' not found. Run a NuGet package restore to generate this file. [.../src/roslyn/src/Compilers/Server/VBCSCompiler/VBCSCompiler.csproj]
.../Tools/dotnetcli/sdk/2.1.101/Sdks/Microsoft.NET.Sdk/build/Microsoft.PackageDependencyResolution.targets(167,5): error : Assets file '.../src/roslyn/Binaries/Obj/VBCSCompiler/project.assets.json' not found. Run a NuGet package restore to generate this file. [.../src/roslyn/src/Compilers/Server/VBCSCompiler/VBCSCompiler.csproj]
```

Adding `VBCSCompiler` to the source-build solution fixes that.

Reply:
Agreed, I actually think we should start putting links to issues in the patch files themselves. I'll do that here.

(Technically Roslyn does have the SDK version change patch still, but I understand that doesn't really count because there isn't much that can be done in Roslyn itself to remove it. We're handling that source-build-side with https://github.com/dotnet/source-build/issues/424.)

Reply:
agree issues + links in the patch file. 

As for SDK we should be good once we move to the next LKG. Then we should be patch-clean++.